wsmaster/che-core-api-system/src/main/java/org/eclipse/che/api/system/server/ServiceTerminator.java

```diff
@@ -44,16 +54,110 @@
   void terminateAll() throws InterruptedException {
     for (ServiceTermination termination : terminations) {
       LOG.info("Shutting down '{}' service", termination.getServiceName());
-      eventService.publish(new StoppingSystemServiceEvent(termination.getServiceName()));
+      doTerminate(termination);
+    }
+  }
+
+  /**
+   * Suspends system services.
+   *
+   * @throws InterruptedException when suspending is interrupted
+   */
+  void suspendAll() throws InterruptedException {
+    for (ServiceTermination termination : terminations) {
+      LOG.info("Suspending down '{}' service", termination.getServiceName());
+      eventService.publish(new SuspendingSystemServiceEvent(termination.getServiceName()));
       try {
-        termination.terminate();
+        termination.suspend();
+      } catch (UnsupportedOperationException e) {
+        LOG.info(
+            "Suspending down '{}' service isn't supported, terminating it",
+            termination.getServiceName());
+        doTerminate(termination);
```

Comment:
@mshaposhnik  Is it OK to publish SystemServiceSuspendedEvent and SystemServiceStoppedEvent at the same time if suspending is not supported?

Reply:
Will move suspended event and log into `try {}` block